### PR TITLE
initial Commit

### PR DESCRIPTION
**Pull request by JustinJohnston-alt**
## Description

Related issue: [990](https://dvsa.atlassian.net/browse/MOTRECALLS-990)

## Before submitting (or marking as "ready for review")

- [ x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [ x] Have you performed a self-review of the code
- [ x] Have you have added tests that prove the fix or feature is effective and working
- [ x] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: src/handler/cloudWatchEvent.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const handler = async (event: ScheduledEvent): Promise<Record<string, number | string>> => {
+  logger.info('Cloudwatch event successfully triggered!');
+
+  return {
+    statusCode: 200,
+    body: `Cloudwatch event ${event.id} successfully triggered!`,
+  };
+};
```
Comment (bennetts101) [inline src/handler/cloudWatchEvent.ts]: do we require this for anything? if not then please delete

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Here's a [handy resource](https://github.com/alphagov/styleguides/blob/master/git.md) for writing good commit messages and general git housekeeping.
+
+Once you have committed your changes, raise a pull request into master. Add as much information as you can to the pull request, make sure you complete the template and to include the issue number you're changes are addressing.
+
+Before raising a pull request, ensure all tests pass and you have run the linting tools. See the README for instructions on how to do this.
+
+## Need help?
+
+Contact Shaun Hare or Anna Dodson for help with this project.
```
Comment (bennetts101) [inline CONTRIBUTING.md]: Was we asked to include Anna if not suggest remove her name.

File: Postman/Recalls.postman_collection.json [hunk 3]
```diff
@@ -0,0 +1,988 @@
...
...tpZCI6IjJaUXBKM1VwYmpBWVhZR2FYRUpsOGxWMFRPSSJ9.eyJhdWQiOiJodHRwczovL2R2c2F0ZXN0LWNsb3VkLnVrLzRhZGUwMzFkLTZlZjEtNGZiOC1hODczLTVkNjU1MmZlMDZkNCIsImlzcyI6Imh0dHBzOi8vc3RzLndpbmRvd3MubmV0LzZjNDQ4ZDkwLTRjYTEtNGNhZi1hYjU5LTBhMmFhNjdkNzgwMS8iLCJpYXQiOjE2NjkyODg0ODEsIm5iZiI6MTY2OTI4ODQ4MSwiZXhwIjoxNjY5Mjg5OTgxLCJhaW8iOiJFMlpnWUdqNzQ1OTAwbjJWYkx2Zm9aU1d2KzhqQVE9PSIsImFwcGlkIjoiY2M5YmNjZjYtOGM3ZS00OGYzLWI4YmUtZGE1YzVhZTZlNzVhIiwiYXBwaWRhY3IiOiIxIiwiaWRwIjoiaHR0cHM6Ly9zdHMud2luZG93cy5uZXQvNmM0NDhkOTAtNGNhMS00Y2FmLWFiNTktMGEyYWE2N2Q3ODAxLyIsIm9pZCI6IjU0MDM5ZGQyLTk1MDUtNGZhNC1iNjlhLWNiYTRjMzBlOWYwYiIsInJoIjoiMC5BVWdBa0kxRWJLRk1yMHlyV1FvcXBuMTRBUjBEM2tyeGJyaFBxSE5kWlZMLUJ0UklBQUEuIiwic3ViIjoiNTQwMzlkZDItOTUwNS00ZmE0LWI2OWEtY2JhNGMzMGU5ZjBiIiwidGlkIjoiNmM0NDhkOTAtNGNhMS00Y2FmLWFiNTktMGEyYWE2N2Q3ODAxIiwidXRpIjoiWWlWdGVyY0d3RXFhcjhaeEIwYXBBQSIsInZlciI6IjEuMCJ9.Ok8rv-jEZ4yLX1Q6ruzMO0dXxvOBYd9PhVec1omwkn-DyFeRXVYwmX5-b82fwCtEN9lDs-tdF6kkx63t1EaUKCz6SKYqzfZLMRe361MacPV310JAuAg78m88cAuaq1MUSqG39F6iyUWrW7WW9xuwOJq310WZl8uNpT0SL4Qx-4JCyrNCsXA6QHdF1crk8p0yOku-GfUyhWqSBW7M-5ndF2lO7Tlf5eYEo-P3z3327pAOAjHz9mNuzyiUFXZMjcp0NyROBGeZmbuXZGO9faoAbkBtQpVo0x8_IrHtOa3BUwF4tolEEnIL3y10B3ik-9QGjGkkxkHdjJdfMiBiuRm-KQ",
```
Comment (bennetts101) [inline Postman/Recalls.postman_collection.json]: please replace with generic token

File: Postman/Recalls.postman_collection.json [hunk 4]
```diff
@@ -0,0 +1,988 @@
...
```
Comment (bennetts101) [inline Postman/Recalls.postman_collection.json]: please replace with generic token

File: Postman/Recalls.postman_collection.json [hunk 5]
```diff
@@ -0,0 +1,988 @@
...
```
Comment (bennetts101) [inline Postman/Recalls.postman_collection.json]: please replace with generic token

File: Postman/Recalls.postman_collection.json [hunk 6]
```diff
@@ -0,0 +1,988 @@
...
```
Comment (bennetts101) [inline Postman/Recalls.postman_collection.json]: please replace with generic token

File: README copy.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
+# Recalls API tester
+
+A starting pattern for AWS lambda in Typescript
```
Comment (bennetts101) [inline README copy.md]: Is this description correct?

File: README copy.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
+# Recalls API tester
```
Comment (bennetts101) [inline README copy.md]: should file be called README copy.md?

File: README copy.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+- `npm run build:dev`
+
+### Run using sam local
+Ensure build steps have been completed.
+
+1. `npm run start:dev`
+1. The lambdas should be available at `http://localhost:3000/`
+
+###  To run recalls API tester locally
+To run the recalls faker api locally, you will need to use Postman to send requests at `http://localhost:3000/`. A [postman collection](Postman/Recalls.postman_collection.json) has been included with this repo.
```
Comment (bennetts101) [inline README copy.md]: should this say mock api rather than faker?

File: package.json [hunk 10]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "dvsa-recalls-api-tester",
```
Comment (bennetts101) [inline package.json]: dvsa-recalls-mock-api ?

File: package.json [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+  "version": "1.0.0",
+  "description": "A package to act as a starter for typescript lambdas",
```
Comment (bennetts101) [inline package.json]: generic description ...

File: package.json [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+    "start:dev": "sam local start-api",
+    "invoke": "sam local invoke"
+  },
+  "contributors": [
+    {
+      "name": "Shaun Hare",
+      "email": "shaun.hare@dvsa.gov.uk"
+    },    {
+      "name": "Justin Johnston",
+      "email": "justin.johnston@kainos.com"
```
Comment (bennetts101) [inline package.json]: suggest just leave Shaun Hare?

File: recalls-external-api-specification-0.3.0.yml [hunk 13]
```diff
@@ -0,0 +1,302 @@
+---
+openapi: 3.0.1
+info:
+  title: DVSA - Vehicle Recalls Service API
+  description: API to allow vehicle manufacturers to manage recording outstanding safety recalls for vehicles with DVSA
+  version: 0.3.0
```
Comment (bennetts101) [inline recalls-external-api-specification-0.3.0.yml]: please update to 0.3.1

File: src/handler/auth.ts [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+import HttpResponse from '../util/httpResponse';
+import {
+  ErrorMissingGrantType, AuthenticatedMessage, ErrorMissingClientID, ErrorMissingClientSecret, ErrorMissingScope,
+} from '../util/authorisationResponses';
+import {
+  containsClientId, containsClientSecret, containsGrantType, containsScope,
+} from '../util/validatorsAuthentication';
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
```
Comment (bennetts101) [inline src/handler/auth.ts:13]: is `async` required if you are disabling require-await?

File: src/handler/delete.ts [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+  }
+  if (!event.pathParameters.vin) {
+    return HttpResponse(StatusCodes.NOT_FOUND, getReasonPhrase(StatusCodes.NOT_FOUND));
+  }
+  if (!event.queryStringParameters.manufacturerCampaignReference && !event.queryStringParameters.dvsaCampaignReference) {
+    return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.DvsaAndManufacturerCampaignReferenceMissing);
+  }
+  const { vin } = event.pathParameters;
+  const { manufacturerCampaignReference, dvsaCampaignReference } = event.queryStringParameters;
+  const vehicles = Vehicles;
```
Comment (bennetts101) [inline src/handler/delete.ts]: why assign to variable with slightly different name?

File: CONTRIBUTING.md (hunk 2 above)
Comment (JustinJohnston-alt) [reply CONTRIBUTING.md]: removed the second name

File: Postman/Recalls.postman_collection.json (hunk 3 above)
Comment (JustinJohnston-alt) [reply Postman/Recalls.postman_collection.json]: replaced with generic token

File: Postman/Recalls.postman_collection.json (hunk 4 above)
Comment (JustinJohnston-alt) [reply Postman/Recalls.postman_collection.json]: replaced with generic token

File: Postman/Recalls.postman_collection.json (hunk 5 above)
Comment (JustinJohnston-alt) [reply Postman/Recalls.postman_collection.json]: replaced with generic token

File: Postman/Recalls.postman_collection.json (hunk 6 above)
Comment (JustinJohnston-alt) [reply Postman/Recalls.postman_collection.json]: replaced with generic token

File: README copy.md (hunk 7 above)
Comment (JustinJohnston-alt) [reply README copy.md]: changed to be more relevant

File: README copy.md (hunk 8 above)
Comment (JustinJohnston-alt) [reply README copy.md]: deleted and updated readme.md

File: README copy.md (hunk 9 above)
Comment (JustinJohnston-alt) [reply README copy.md]: changed

File: package.json (hunk 10 above)
Comment (JustinJohnston-alt) [reply package.json]: I have updated this

File: package.json (hunk 11 above)
Comment (JustinJohnston-alt) [reply package.json]: I have updated this

File: package.json (hunk 12 above)
Comment (JustinJohnston-alt) [reply package.json]: removed my name

File: recalls-external-api-specification-0.3.0.yml (hunk 13 above)
Comment (JustinJohnston-alt) [reply recalls-external-api-specification-0.3.0.yml]: file has been replaced with newer version

File: src/handler/auth.ts (hunk 14 above)
Comment (JustinJohnston-alt) [reply src/handler/auth.ts:13]: I need the async as part of the SAM local chain, this was part of the starter.

File: src/handler/delete.ts (hunk 15 above)
Comment (JustinJohnston-alt) [reply src/handler/delete.ts]: changed this in all 4 functions

File: src/handler/cloudWatchEvent.ts (hunk 1 above)
Comment (JustinJohnston-alt) [reply src/handler/cloudWatchEvent.ts]: deleted the cloudwatch events

File: .gitignore [hunk 16]
```diff
@@ -0,0 +1,108 @@
+# Logs
```
Comment (DariuszJerzewski) [inline .gitignore:1]: Could you add a gitignore for vscode, please?

https://www.toptal.com/developers/gitignore/api/visualstudiocode

**Review by DariuszJerzewski — APPROVED** [review]

File: .vscode/launch.json [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+      "address": "localhost",
+      "port": 5858,
+      "localRoot": "${workspaceRoot}/.aws-sam/build/PostLambdaFunction",
+      "remoteRoot": "/var/task",
+      "protocol": "inspector",
+      "stopOnEntry": false,
+      "sourceMaps": true
+    },
+    {
+      "name": "CloudWatchEventLambdaFunction",
```
Comment (bennetts101) [inline .vscode/launch.json]: I think this can also be removed now?

File: .gitignore (hunk 16 above)
Comment (JustinJohnston-alt) [reply .gitignore:1]: done

File: .vscode/launch.json (hunk 17 above)
Comment (JustinJohnston-alt) [reply .vscode/launch.json]: removed

File: package.json [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+  "description": "A SAM Local application to help development teams test output from their solution locally for the safety recalls API",
+  "main": "index.js",
+  "scripts": {
+    "lint": "eslint '*/**/*.ts' --quiet --fix",
+    "lint:ci": "eslint '*/**/*.ts'",
+    "test": "jest --coverage",
+    "test:ci": "jest --ci",
+    "build:dev": "NODE_ENV=development webpack-cli --config webpack/webpack.development.js",
+    "watch:dev": "NODE_ENV=development webpack-cli --config webpack/webpack.development.watch.js",
+    "build:prod": "webpack-cli --config webpack/webpack.production.js",
```
Comment (bennetts101) [inline package.json:13]: do we need a prod build for anything, if not suggest remove this

File: package.json [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+    "ts-loader": "^9.2.6",
+    "typescript": "^4.4.3",
+    "uuid": "^8.3.2",
+    "webpack": "^5.56.1",
+    "webpack-cli": "^4.8.0",
+    "webpack-merge": "^5.8.0",
+    "winston": "^3.6.0"
+  },
+  "engines": {
+    "node": "^16.17.1"
```
Comment (bennetts101) [inline package.json]: can we make this 14.7.3 please to align with recalls api

File: package.json [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+    "@types/jest": "^28.1.4",
+    "@types/node": "^16.10.2",
+    "@types/uuid": "^8.3.1",
+    "archiver": "^5.3.0",
+    "aws-sam-webpack-plugin": "^0.12.0",
+    "copy-webpack-plugin": "^11.0.0",
+    "current-git-branch": "^1.1.0",
+    "dotenv": "^16.0.0",
+    "eslint": "^8.23.0",
+    "eslint-import-resolver-typescript": "^3.5.0",
```
Comment (bennetts101) [inline package.json]: can we make this 2.7.1 please to align with node engine

File: webpack/webpack.production.js [hunk 21]
```diff
@@ -0,0 +1,74 @@
+const fs = require('fs-extra')
```
Comment (bennetts101) [inline webpack/webpack.production.js:1]: can we remove this file?

File: webpack.config.js [hunk 22]
```diff
@@ -0,0 +1,44 @@
+const path = require('path');
```
Comment (bennetts101) [inline webpack.config.js:1]: can we remove this file?

**Comment by bennetts101** [comment]
can't leave comment on file but is package-lock up to date?

**Pull request merged**